ProcessLib/BoundaryCondition/GenericNonuniformNaturalBoundaryCondition-impl.h

```diff
@@ -0,0 +1,94 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "GenericNonuniformNaturalBoundaryCondition.h"
+#include "GenericNonuniformNaturalBoundaryConditionLocalAssembler.h"
+#include "MeshLib/MeshSearch/NodeSearch.h"
+#include "ProcessLib/Utils/CreateLocalAssemblers.h"
+
+namespace ProcessLib
+{
+template <typename BoundaryConditionData,
+          template <typename, typename, unsigned>
+          class LocalAssemblerImplementation>
+template <typename Data>
+GenericNonuniformNaturalBoundaryCondition<BoundaryConditionData,
+                                          LocalAssemblerImplementation>::
+    GenericNonuniformNaturalBoundaryCondition(
+        unsigned const integration_order, unsigned const shapefunction_order,
+        unsigned const global_dim,
+        std::unique_ptr<MeshLib::Mesh>&& boundary_mesh, Data&& data)
+    : _data(std::forward<Data>(data)), _boundary_mesh(std::move(boundary_mesh))
+{
+    static_assert(std::is_same<typename std::decay<BoundaryConditionData>::type,
+                               typename std::decay<Data>::type>::value,
+                  "Type mismatch between declared and passed BC data.");
+
+#if 0
+    // TODO fix/improve check!
```

Comment:
Maybe we enable the non-complete check anyway and keep the TODO improve check (how?) for the time being?